app/src/main/java/com/example/healthinspector/Fragments/UserProfileFragment.java

```diff
@@ -1,35 +1,107 @@
+            }
+        });
+        binding.addAllergyImageView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                FragmentTransaction fragmentTransaction =  getActivity().getSupportFragmentManager().beginTransaction();
+                fragmentTransaction.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN);
+
+                searchFragment = new SearchFragment();
+                Bundle bundle = new Bundle();
+                bundle.putSerializable(Constants.PREVIOUS_FRAGMENT, SearchFragmentSwitch.ALLERGEN_SEARCH);
+                searchFragment.setArguments(bundle);
+                fragmentTransaction.replace(R.id.fragment_container, searchFragment);
+                fragmentTransaction.addToBackStack(null);
+                fragmentTransaction.commit();
+            }
+        });
+
+        binding.addWarningImageView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                FragmentTransaction fragmentTransaction =  getActivity().getSupportFragmentManager().beginTransaction();
+                fragmentTransaction.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN);
+
+                searchFragment = new SearchFragment();
+                Bundle bundle = new Bundle();
+                bundle.putSerializable(Constants.PREVIOUS_FRAGMENT, SearchFragmentSwitch.ADDITIVE_SEARCH);
+                searchFragment.setArguments(bundle);
+                fragmentTransaction.replace(R.id.fragment_container, searchFragment);
+                fragmentTransaction.addToBackStack(null);
+                fragmentTransaction.commit();
+            }
+        });
+
         return view;
     }
 
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
+
+        ArrayList<String> userAllergies = (ArrayList) ParseUser.getCurrentUser().get("userAllergies");
+        ItemAdapter allergiesAdapter = new ItemAdapter(requireContext(), userAllergies, SearchFragmentSwitch.ADDITIVE_SEARCH);
+        binding.userAllergiesRecyclerView.setAdapter(allergiesAdapter);
+        LinearLayoutManager linearLayoutManagerAllergies = new LinearLayoutManager(getContext());
+        binding.userAllergiesRecyclerView.setLayoutManager(linearLayoutManagerAllergies);
+
+        ArrayList<String> userWarnings = (ArrayList) ParseUser.getCurrentUser().get("userWarningIngredients");
+        ItemAdapter warningsAdapter = new ItemAdapter(requireContext(), userWarnings, SearchFragmentSwitch.ADDITIVE_SEARCH);
+        binding.userAllergiesRecyclerView.setAdapter(warningsAdapter);
+        LinearLayoutManager linearLayoutManagerWarnings = new LinearLayoutManager(getContext());
+        binding.userAllergiesRecyclerView.setLayoutManager(linearLayoutManagerWarnings);
     }
```

Comment:
Similarly here, the only thing different is "userAllergies" vs "userWarningIngredients"
We can pass those into a generic function that handles the redundant code